openspp-project/openspp-registry
# Add Automatic Approval for OpenSPP Self-Reg Change Requests

## **Why is this change needed?**
A new function in the Change Request module is needed to support the direct approval and application of CRs submitted by the **OpenSPP Self-reg app**. The following will be provided by the new function:

- Accept CRs only from users belonging to the "**Change Request External API**" group.
- Validate the CR input data.
- Skip the validation of required attachments.
- Automatically approve the CR after the input data has been validated.

## **How was the change implemented?**

- A new function "**approve_cr**" was added to the **spp.change.request** model.
- This function can be called by external APIs (XMLRPC) authenticated with a user belonging to the "**Change Request External API**" group.

## **New unit tests**
None
## **Unit tests executed by the author**
None
## **How to test manually**

- Use the OpenSPP Self-reg app to create a new change request.
- Check the new change request in the OpenSPP change request UI.
- Its status must be automatically set to "Applied". 

## **Related links**
[Add Automatic Approval for OpenSPP Self-Reg Change Requests](https://github.com/OpenSPP/openspp-modules/issues/635)
